### PR TITLE
fix inflate crash on Android 10 by upgrading to iconics 4.0.2 (still crashes on Android 9)

### LINKED ISSUE #2955: Phimpme
WhatsApp Image 2019-12-30 at 2 06 25 PM (1)

Phimpme is not working on my phone it has crashed. I have sent feedback as well here is the crash url
https://del.dog/iferfikibe

### LINKED ISSUE #2966: App Not Responding
When I attempt to open the app on my google pixel 3a, the app states that it is unable to respond and it keeps stopping. I don't know if this is a problem on my device, but even after I redownload it, it still gives mew the same error message.

### LINKED ISSUE #2977: The App is not starting in Pixel costom Q
https://del.dog/amamurecyv.txt

Link of Log

### LINKED ISSUE #2982: Crash on application start
**Describe the bug**
I forked [phimpme-android](https://github.com/fossasia/phimpme-android) development branch to fix it because on my device it crashes on startup. The fix was migrating to iconics 4.0.2
The migrated app can be found here: [phimpme fork](https://github.com/TBog/phimpme-android/commits/tbog/iconics)
Now it still crashes on app start-up inside `gallery\data\providers\MediaStoreProvider.java` in method 
`public static ArrayList<Album> getAlbums(Context context)`

**To Reproduce**
Upon app start the app closes instantly.
Seems like the query has changed and the GROUP BY 'hack' used is not longer working.

**Logs**
```
2020-01-17 10:13:30.096 9755-10705/? E/DatabaseUtils: Writing exception to parcel
    android.database.sqlite.SQLiteException: near "GROUP": syntax error (code 1 SQLITE_ERROR[1]): , while compiling: SELECT parent, bucket_display_name FROM files WHERE ((_data>'') AND (owner_package_name IN ( 'org.fossasia.phimpme' ) OR media_type=2 OR media_type=4 OR media_type=0 AND mime_type LIKE 'audio/%' OR media_type=3 OR media_type=0 AND mime_type LIKE 'video/%' OR media_type=1 OR media_type=0 AND mime_type LIKE 'image/%') AND (is_pending=0) AND (is_trashed=0) AND (volume_name IN ( 'external_primary' ))) AND ((media_type=? ) GROUP BY ( parent ))
        at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
        at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:1364)
        at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:803)
        at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:590)
        at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:62)
        at android.database.sqlite.SQLiteQuery.<init>(SQLiteQuery.java:37)
        at android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:46)
        at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1959)
        at android.database.sqlite.SQLiteQueryBuilder.query(SQLiteQueryBuilder.java:604)
        at com.android.providers.media.MediaProvider.queryInternal(MediaProvider.java:2167)
        at com.android.providers.media.MediaProvider.query(MediaProvider.java:1978)
        at android.content.ContentProvider$Transport.query(ContentProvider.java:267)
        at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:104)
        at android.os.Binder.execTransactInternal(Binder.java:1021)
        at android.os.Binder.execTransact(Binder.java:994)
```

**Smartphone Info:**

|               | |
|---------------|-|
|Device         | Samsung Galaxy S10 plus |
|Android Version| Android version 10    |

**Would you like to work on the issue?**

- [x] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by TBog**
Fixes #2982
Should fix #2977
Should fix #2966
Should fix #2955

Make the app work on Android 10
Some iconics icons got lost, please help fix them ;)

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -154,8 +154,8 @@ dependencies {
     implementation "androidx.vectordrawable:vectordrawable:1.1.0"
     implementation "androidx.palette:palette:1.0.0"
     implementation "androidx.cardview:cardview:1.0.0"
-    implementation "androidx.browser:browser:1.0.0"
-    implementation "androidx.recyclerview:recyclerview:1.0.0"
+    implementation "androidx.browser:browser:1.2.0"
```
Comment (codedsun) [inline app/build.gradle]: Revert

File: app/build.gradle [hunk 2]
```diff
@@ -154,8 +154,8 @@ dependencies {
...
+    implementation "androidx.recyclerview:recyclerview:1.1.0"
```
Comment (codedsun) [inline app/build.gradle]: Revert

File: app/build.gradle [hunk 3]
```diff
@@ -224,7 +225,7 @@ dependencies {
 
     //tests
     testImplementation "junit:junit:$rootProject.junitVersion"
-    androidTestImplementation 'com.google.code.findbugs:jsr305:3.0.0'
+    androidTestImplementation 'com.google.code.findbugs:jsr305:3.0.2'
```
Comment (codedsun) [inline app/build.gradle]: revert

**Comment by codedsun** [comment]
@TBog - Also the CI is failing please fix and if this works, let's do this quickly.

**Comment by TBog** [comment]
```
Execution failed for task ':app:spotlessJava'.
> The following files had format violations
```
- Where can I get the formatting rules?
- This PR was a quick fix mostly for myself, that's why I upgraded some libraries, I'll revert them on Monday
- Some iconics icons can no longer be found and I replaced them with some random ones that had similar names, this is not a proper fix, please help me with this as I don't know what they looked like before

**Comment by codedsun** [comment]
> ```
> Execution failed for task ':app:spotlessJava'.
> > The following files had format violations
> ```
> 
> 
> 
> * Where can I get the formatting rules?

run `./gradlew spotlessApply` in the cmd

> * This PR was a quick fix mostly for myself, that's why I upgraded some libraries, I'll revert them on Monday

Ok, please upload a gif as well.


> * Some iconics icons can no longer be found and I replaced them with some random ones that had similar names, this is not a proper fix, please help me with this as I don't know what they looked like before

Yes, sure please point where you need it I will help you for that.

**Comment by TBog** [comment]
> I also tried but was getting some layout inflater error.

What error? Can you post a log? Does it crash?

**Comment by codedsun** [comment]
@TBog - https://github.com/mikepenz/Android-Iconics/issues/480 The link of the issue which I got. 
Please let's not waste time in debugging this when you got this working. Please just finalize this PR fast so that we can make a release. if their's an issue do mention me.

**Comment by TBog** [comment]
|  iconics icon that can no longer be found | my replacement |
| --- | --- |
| CommunityMaterial.Icon.cmd_check_circle | CommunityMaterial.Icon.cmd_send_check |
| CommunityMaterial.Icon.cmd_check | CommunityMaterial.Icon.cmd_select_marker |
| CommunityMaterial.Icon.cmd_arrow_left | CommunityMaterial.Icon.cmd_subdirectory_arrow_left |
| CommunityMaterial.Icon.cmd_check_circle | GoogleMaterial.Icon.gmd_check_circle |
| CommunityMaterial.Icon.cmd_check | GoogleMaterial.Icon.gmd_check |
| CommunityMaterial.Icon.cmd_folder | GoogleMaterial.Icon.gmd_folder |
| CommunityMaterial.Icon.cmd_compass_outline | CommunityMaterial.Icon.cmd_ruler_square_compass |

**Comment by TBog** [comment]
changed `app:iiv_icon="faw_trash_o"` to `app:iiv_icon="gmd_delete"`
here is the gif
small20fps

**Comment by TBog** [comment]
> Please let's not waste time in debugging this when you got this working. Please just finalize this PR fast so that we can make a release. if their's an issue do mention me.

@codedsun I fixed the crash you had by commenting out `attachBaseContext` from `org/fossasia/phimpme/base/ThemedActivity.java`

**Comment by codedsun** [comment]
@TBog - You mentioned earlier that your app wasn't crashing without fixing the inflate crash

**Comment by TBog** [comment]
> You mentioned earlier that your app wasn't crashing without fixing the inflate crash

On Android 10 this PR fixes the startup crash, but on Android 9 it still crashes without ad23847d8e4fda7d09150fb6804baa06ac548dab

**Comment by TBog** [comment]
This PR does not fix the crash on Android 9 so I'm closing it in favor of #2987

**Pull request closed without merging**